src/index.c

```diff
@@ -746,6 +751,26 @@ static int truncate_racily_clean(git_index *index)
 	return 0;
 }
 
+unsigned git_index_version(git_index *index)
+{
+	assert(index);
+
+	return index->version;
+}
+
+int git_index_set_version(git_index *index, unsigned int version)
+{
+	assert(index);
+
+	if (version < INDEX_VERSION_NUMBER_LB ||
+		version > INDEX_VERSION_NUMBER_UB)
+		return -1;
```

Comment:
Please `giterr_set` a specific error message here.  Eg, `giterr_set(GITERR_INDEX, "Invalid version number")`.

If this is wildly bogus data that the caller is feeding us, you can also feel free to assert here instead.